pandas/tests/frame/methods/test_explode.py

```diff
@@ -9,14 +9,61 @@ def test_error():
     df = pd.DataFrame(
         {"A": pd.Series([[0, 1, 2], np.nan, [], (3, 4)], index=list("abcd")), "B": 1}
     )
-    with pytest.raises(ValueError, match="column must be a scalar"):
+    with pytest.raises(
+        ValueError, match="column must be a scalar, tuple, or list thereof"
+    ):
+        df.explode([list("AA")])
+
+    with pytest.raises(ValueError, match="column must be unique"):
         df.explode(list("AA"))
 
     df.columns = list("AA")
     with pytest.raises(ValueError, match="columns must be unique"):
         df.explode("A")
 
 
+@pytest.mark.parametrize(
+    "input_dict, input_index, input_subset, error_message",
+    [
+        (
+            {
+                "A": [[0, 1, 2], np.nan, [], (3, 4)],
+                "B": 1,
+                "C": [["a", "b", "c"], "foo", [], ["d", "e", "f"]],
+            },
+            list("abcd"),
+            list("AC"),
+            "columns must have matching element counts",
+        ),
+        (
+            {
+                "A": [[0, 1, 2], np.nan, [], (3, 4)],
+                "B": 1,
+                "C": [["a", "b", "c"], "foo", [], ["d", "e", "f"]],
+            },
+            list("abcd"),
+            [],
+            "column must be nonempty",
+        ),
+        (
+            {
+                "A": [[0, 1, 2], np.nan, [], (3, 4)],
+                "B": 1,
+                "C": [["a", "b", "c"], "foo", [], "d"],
+            },
+            list("abcd"),
+            list("AC"),
+            "columns must have matching element counts",
+        ),
+    ],
+)
+def test_error_multi_columns(input_dict, input_index, input_subset, error_message):
```

Comment:
Can you parametrize only the changing parts?

Reply:
This test_error_multi_columns() is a new function I added in my PR, not something that already exists there. I added one more test case to this test_error_multi_columns() in this commit.

Reply:
That was not what I was referring to. Most of your parametrization values are the same for all cases, so no need to put them into the parametrization

Reply:
@phofl updated. btw "Python Dev / actions-310-dev" fails coverage upload, but that's not caused by me..